tests.py

```diff
@@ -1,81 +1,241 @@
-
-
-# @pytest.mark.parametrize("x, y, z", [(3,4,6), (2,4,6)])
-
-DUMMY_INPUTS = {
-    'initial_hangman': 'fdgfdgd____', 'solution': "sfdjkshdfjsd", 'guesses_allowed': 6,
-    'guesses': ['f', 'd', '3']
-}
+from GameMethod import DummyMethod, HTTPSMethod, EmptyMethod, GameMethod
+from GameUI import TerminalUI, TestUI
+import requests
 
+# CONSTANTS
 INSTRUCTIONS = "For guessing a letter enter one, for a hint enter H, for the solution enter S" \
                "\n "
 WELCOME_MESSAGE = "Hello and Welcome to the best hangman game ever. \n"
 VALID_PATTERN = "[a-z]{1}|H|S"
 BYE_MESSAGE = "Game Over "
-
 UI_OUTPUTS = [WELCOME_MESSAGE, INSTRUCTIONS, VALID_PATTERN, BYE_MESSAGE]
 
-# API TESTS
-"""
-def test_valid_login(url):
-    response = requests.post(url, data='text=great').json()
-    assert response.status_code == 200
-
-def test_valid_get_hint(url):
-    response = requests.get(url, data='text=great').json()
-    assert response.status_code == 200
-
-def test_valid_guess(url):
-    response = requests.put(url, data='text=great').json()
-    assert response.status_code == 200
-
-def test_valid_get_solution(url):
-    response = requests.get(url, data='text=great').json()
-    assert response.status_code == 200
-"""
-
 EMPTY_SUCCESS = "Hello and Welcome to the best hangman game ever. \n\nGame Over \n"
-DUMMY_SUCCESS = ""
-HTTPS_SUCCESS = ""
-SUCCESSES = [EMPTY_SUCCESS, DUMMY_SUCCESS, HTTPS_SUCCESS]
-
-
-def test_integration():
-    test_ui = TerminalUI()
-    test_methods = [EmptyMethod()]
-    # , DummyMethod(), HTTPSMethod()]
-    test_games = [Game(test_ui, test_method) for test_method in test_methods]
-
-    for test_index, test_game in enumerate(test_games):
-        proc = subprocess.Popen(['python', '/Users/kbenarie/Projects/hangman-exercise/main.py'],
-                                stdout=subprocess.PIPE)
-        stdout = proc.stdout.read().decode('utf-8')
-        assert stdout == SUCCESSES[test_index], '{} case did not succeed'.format(
-            SUCCESSES[test_index])
 
 
-def test_guess_letter():
-    letters = ['a', 'd', 't', 'g', 'r']
+# FUNCTIONS
+def validate_JSON(json_data):
+    """
+    This function validates json format on a given data structure
+    :param json_data:
+    :return: True if it is, False otherwise
+    """
+    try:
+        json.loads(json_data)
+    except ValueError as err:
+        return False
+    return True
+
+
+def edit_distance(string1, string2):
+    """
+    This function checks the length of the edit distance between two given strings
+    or in other words - how many operations should i do in order to get from string1 to string2.
+    Assumptions: The strings in this project are necessarily in the same length.
+    :param string1:
+    :param string2:
+    :return: the distance
+    """
+    count_diffs = 0
+    for a, b in zip(string1, string2):
+        if a != b:
+            count_diffs += 1
+    return count_diffs
+
+
+# CLASSES
+
+class TestClassCheckAPI(unittest.TestCase):
```

Comment:
There is no need to mention the word "class" in the name of the class, as it is already declared with the keyword "class". The prefix "Test" is enough. Example: `TestCheckAPI`.
Another thing - "CheckAPI" is somewhat generic. Maybe we can find a more specific name that will help us understand which API is under test, or what are the behaviors that we want to validate?

Reply:
Was changed to 'TestHTTPSContract'